src/main/java/pl/javastart/recipeapp/recipe/RecipeService.java

```diff
@@ -0,0 +1,77 @@
+package pl.javastart.recipeapp.recipe;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import pl.javastart.recipeapp.category.Category;
+import pl.javastart.recipeapp.category.CategoryRepository;
+
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class RecipeService {
+
+    private final RecipeRepository recipeRepository;
+    private final CategoryRepository categoryRepository;
+
+    public RecipeService(RecipeRepository recipeRepository, CategoryRepository categoryRepository) {
+        this.recipeRepository = recipeRepository;
+        this.categoryRepository = categoryRepository;
+    }
+
+    public List<Recipe>find3MostLikeableRecipes() {
+        return recipeRepository.findTop3ByOrderByLikesDesc();
+    }
+
+    public List<Recipe> findAll(){
+        return recipeRepository.findAll();
+    }
+
+    public Recipe findById(Long id){
+        return recipeRepository.findById(id).orElseThrow();
+    }
```

Comment:
Aktualnie wejście na przepis, który nie istnieje powoduje wyświetlenie stacktrace błedu w przeglądarce (/app/recipe?id=100). Lepiej rzucić tutaj dedykowany wyjątek, albo zwrócić Optional<Recipe> i odpowiednio to obsłużyc w kontrolerze